src/dscanner/stats.d

```diff
@@ -52,3 +52,155 @@ ulong printLineCount(Tokens)(File output, string fileName, ref Tokens tokens)
 	output.writefln("%s:\t%d", fileName, count);
 	return count;
 }
+
+void printTokenDump(string fileName)
+{
+	import dmd.tokens;
+	import dmd.lexer;
+	import std.file;
+
+	auto input = readText(fileName);
+	scope lexer = new Lexer(null, input.ptr, 0, input.length, 0, 0);
+    
+	lexer.nextToken;
+	Token t = lexer.token;
+
+	writeln("text                    \tblank\tindex\tline\tcolumn\ttype\tcomment\ttrailingComment");
+	do
+    {
+		import std.stdio : writefln;
+		t = lexer.token;
+		string aux;
+		bool empty = true;
+
+		switch (t.value)
+		{
+			case TOK.string_:
+				aux = t.ustring[0 .. t.len].dup;
+				aux = "\"" ~ aux ~ "\"";
+				writef("<<%20s>>", aux);
+				empty = false;
+				break;
+			case TOK.int32Literal:
+				writef("<<%20d>>", t.intvalue);
+				empty = false;
```

Comment:
It seems that empty should be just set to true if the default case is taken. You would have less lines of code that way.